docs(backup/s3): add readme content (#10233)
camunda/zeebe: backup-stores/s3/README.md

[MinIO]: https://min.io/product/s3-compatibility

## Configuration

**Required**

- `bucketName` Name of the bucket that will be used for storing backups. The bucket must already
  exist.

**Optional**

- `endpoint` URL to use when connecting to S3. If none is provided, the AWS SDK will use a default
  value.
- `region` Region to use when connecting to S3. If none is provided, the AWS SDK will try to
  determine an appropriate value. [^region]
- `credentials` A pair of `accessKey` and `secretKey`. If none are provided, the AWS SDK will try to
  determine an appropriate value. [^credentials]

[^region]: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html#automatically-determine-the-aws-region-from-the-environment
[^credentials]: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html#credentials-chain

## Known limitations

* There is no concept of backup rotation so backups will accumulate unless manually deleted.
* Individual files in a backup may not exceed 5GiB, otherwise the upload may fail.
* If the backup consists of more than 1000 files, deleting a backup may leave behind some objects.

## Internals

> A backup consists of 2 objects used for managing the backup and the objects
> containing the backup data itself.
> All object keys are prefixed by `partitionId/checkpointId/nodeId` which uniquely identifies a
> backup:

Reviewer: ```suggestion
A backup consists of following objects prefixed by `partitionId/checkpointId/nodeId/` :
```

* `metadata.json`: A _metadata_ object, containing metadata serialized as JSON
* `status.json`: A _status_ object, containing the current status serialized as JSON

> * `snapshots/*` Objects for _snapshot files_, additionally prefixed with 'snapshot'

Reviewer: ```suggestion
* `snapshots/*` Objects for _snapshot files_'
```

> * `segments/*` Objects for _segment files_, additionally prefixed with 'segments'

Reviewer: ```suggestion
* `segments/*` Objects for _segment files_
```
